### PR TITLE
[SIRo] FP and Evaluation Fixes

### PR DESCRIPTION
**Pull request by ASzot**

### CODE REVIEW

File: habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml
Comment (xavierpuigf) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml:1]: Are we planning on using oracle_nav_with_backup? Seems like it is not defined here:
@jimmytyyang @ASzot

File: habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot_fp.yaml [hunk 1]
```diff
@@ -46,32 +48,30 @@ habitat:
         motion_control: human_joints
         spawn_max_dist_to_obj: -1
   environment:
-    max_episode_steps: 5000
+    max_episode_steps: 10
```
Comment (xavierpuigf) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot_fp.yaml:51]: change this

File: habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml
Comment (jimmytyyang) [reply habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml:1]: oracle_nav_with_backup allows the backing-up motion to reduce collision in cramped FP scenes

Comment (akshararai) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml:1]: @akshararai  need to fix this config before merge

File: habitat-baselines/habitat_baselines/config/experiments_hab3/single_agent_kinematic_oracle_spot.yaml [hunk 2]
```diff
@@ -13,25 +14,113 @@ hydra:
...
         spawn_max_dist_to_obj: -1.0
+        # For noncylinder navmesh action
+        # If we allow the agent to do sliding
+        allow_dyn_slide: True
+        # If the agent is allowed to slide if the agent needs to do rotation
+        enable_rotation_check_for_dyn_slide: False
+        # The distance threshold of treating collision
+        collision_threshold: 0.02
+        # The navmesh offset for the action control
+        navmesh_offset: [[0.0, 0.0], [0.25, 0.0], [-0.25, 0.0]]
```
Comment (akshararai) [inline habitat-baselines/habitat_baselines/config/experiments_hab3/single_agent_kinematic_oracle_spot.yaml:117]: @akshararai update to the new offsets

File: habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py [hunk 3]
```diff
@@ -234,12 +234,23 @@ def _sample_active_idxs(self) -> Tuple[np.ndarray, np.ndarray]:
...
                     "The current code only supports sampling one agent of a given type at a time"
                 )
-            active_agents_type = self._rnd.choice(
-                self._agent_count_idxs[agent_type_ind],
-                size=self._pop_config.num_active_agents_per_type[
-                    agent_type_ind
-                ],
-            )
+            if (
+                self._pop_config.force_partner_sample_idx >= 0
```
Comment (akshararai) [inline habitat-baselines/habitat_baselines/rl/multi_agent/multi_agent_access_mgr.py:238]: flagging that we call _pop_config many times in this piece of code. Might be better to 
`force_partner_sample_idx = self._pop_config.force_partner_sample_idx ` at the very start.

File: habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_fp.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# @package _global_
+
+defaults:
+  - /habitat: habitat_config_base
+  - /habitat/simulator/agents@habitat.simulator.agents.main_agent: depth_head_agent
+  - /habitat/task/rearrange: rearrange_easy_fp
+  - /habitat/dataset/rearrangement: replica_cad
```
Comment (akshararai) [inline habitat-lab/habitat/config/benchmark/rearrange/rearrange_easy_fp.yaml:7]: should this dataset be fp?

File: habitat-lab/habitat/config/habitat/task/rearrange/multi_agent_tidy_house_base_spot.yaml [hunk 5]
```diff
@@ -39,8 +39,11 @@ constraint_violation_drops_object: True
 task_spec: multi_agent_tidy_house_fp
 pddl_domain_def: fp
 actions:
-  # Actions are defined per agent. Unlike the sensors, actions are not
-  # automatically duplicated per agent.
+  agent_0_base_velocity:
+    agent_index: 0
+    lin_speed: 40.0
```
Comment (akshararai) [inline habitat-lab/habitat/config/habitat/task/rearrange/multi_agent_tidy_house_base_spot.yaml:44]: Why are setting agent_0_base_velocity? Sorry a little confused, I thought we were only using oracle nav? Is this still correct?

File: habitat-lab/habitat/config/habitat/task/rearrange/multi_agent_tidy_house_base_spot.yaml [hunk 6]
```diff
@@ -53,14 +56,18 @@ actions:
...
-    navmesh_offset_for_agent_placement: [[0.0, 0.0], [0.25, 0.0], [-0.25, 0.0]]
+    navmesh_offset_for_agent_placement: [[0.0, 0.0], [0.15, 0.0], [-0.15, 0.0]]
     enable_lateral_move: False
     turn_thresh: 0.1
     dist_thresh: 0.5
+    lateral_lin_speed: 40.0
+    longitudinal_lin_speed: 40.0
   agent_1_base_velocity:
     agent_index : 1
+    lin_speed: 40.0
```
Comment (akshararai) [inline habitat-lab/habitat/config/habitat/task/rearrange/multi_agent_tidy_house_base_spot.yaml:69]: same question. Does this need to be defined twice?

**Review by akshararai — APPROVED** [review]
left some comments assigned to myself. Questions about the base velocity and rearrage_easy_fp benchmark.

File: habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml
Comment (akshararai) [reply habitat-baselines/habitat_baselines/config/experiments_hab3/pop_play_kinematic_oracle_humanoid_spot.yaml:1]: Added

**Pull request merged**